### PR TITLE
E2E (Atomic): Fix comment like tests

### PR DESCRIPTION
**Pull request by WunderBart**
Tracking issue: #65906

### Proposed Changes

Make the following test compatible with the Atomic environment:

> **Post a comment**
> specs/published-content/likes__comment.ts: Likes (Comment)

> **Like the comment**
> specs/published-content/likes__comment.ts: Likes (Comment)

> **Like the comment**
> specs/published-content/likes__comment.ts: Likes (Comment)

### Why was it failing?

There's a big difference between comments+likes on Simple vs. Atomic sites: Simple sites render everything directly, while Atomic via widget iframes, and I need to say that test-wise, iframes are much more problematic. 😅 The first issue was to figure out all the selectors, as even though those components look identical, their implementation is different. The second one was to find out why the (comment) Like button is sometimes freezing at the `Loading…` state. The third problem was the comment form blurring right after being focused, which I was finally able to resolve by working on the final, and the most challenging issue: why the comment form and the like buttons on Atomic sites are sometimes not immediately actionable and why is it not handled by Playwright's built-in auto-waiting mechanism.

Regarding the latter, after hours of debugging, I'm 87.89% sure that the reason for all the flakiness is simply no DOM indication of the moment they become actionable, i.e., the scripts have been fully initialized. For example, the Like button can become perfectly clickable (render-wise) while the script bit is still being initialized. The only indication of that behaviour is that there are still initialization `message` events being dispatched **after** the button has been rendered (like this one fbhepr%2Skers%2Sjcpbz%2Sjvqtrgf.jc.pbz%2Syvxrf%2Syvxrf%2Qerfg.wf%3Se%3Qqs0pn9s4%231174-og). 

For the fix, I've built a custom helper that waits until those `message` events from `https://widgets.wp.com` become idle (similar to Playwright's `networkidle` check, but for events dispatch). This solution is not targeted at a specific widget, but in this global (page-wise) form, I think it will perform even better by waiting for all the widgets to be loaded, which is also when the page's layout stops _jumping_.

### Testing instruction

The fixed tests should pass for both Simple and Atomic sites (production).

### CODE REVIEW

File: packages/calypso-e2e/src/element-helper.ts [hunk 1]
```diff
@@ -152,3 +152,28 @@ export async function getIdFromBlock( block: Locator ): Promise< string > {
...
+export async function waitForWPWidgets( page: Page ): Promise< void > {
+	await page.evaluate( async () => {
+		await new Promise( ( resolve ) => {
+			const wait = () => setTimeout( resolve, 1000 );
+			let timeout = wait();
+
+			window.addEventListener( 'message', ( event ) => {
+				if ( event.origin === 'https://widgets.wp.com' ) {
+					clearTimeout( timeout );
+					timeout = setTimeout( resolve, 1000 );
```
Comment (dpasque) [inline packages/calypso-e2e/src/element-helper.ts]: **Suggestion (non-blocking)**: (Mostly for consistency), I believe you can re-use that `wait` function you've defined above here?

**Review by dpasque — APPROVED** [review]
Wow, nice work on this one! This seems like it was brutal. 😩 These cases where something isn't ready, but not in a DOM-obvious way, are so rough!

Your changes read so nicely, and the tests pass! 🥳

**Comment by worldomonation** [comment]
This change looks good to me and it's amazing just how involved it is to work around our comment like button. Great work!

I have one question that is related to the Likes and Comments spec, but not related to this PR. Would it be beneficial to implement a method in the `RestAPIClient` to add a comment to a post, run the test, then clean up by removing the comment?

**Review by worldomonation — APPROVED** [review]

File: packages/calypso-e2e/src/element-helper.ts (hunk 1 above)
Comment (WunderBart) [reply packages/calypso-e2e/src/element-helper.ts]: Ah, yes! Nice catch 👍

**Comment by WunderBart** [comment]
> This seems like it was brutal. 😩 These cases where something isn't ready, but not in a DOM-obvious way, are so rough!

> (…) it's amazing just how involved it is to work around our comment like button.

Couldn't agree more. This was indeed a tough one, but at the same time it's a valuable lesson for me. From now on, whenever I'm working on a dynamically updated content, I'll be paying extra close attention to make sure the interactability is signalled correctly so the whole thing is easy to test. 💪😅

**Comment by WunderBart** [comment]
> I have one question that is related to the Likes and Comments spec, but not related to this PR. Would it be beneficial to implement a method in the `RestAPIClient` to add a comment to a post, run the test, then clean up by removing the comment?

I haven't worked with the `RestAPIClient` before, but it sounds like it could be a more convenient and reliable way of getting what we actually want to test here! 🙌

File: packages/calypso-e2e/src/element-helper.ts (hunk 1 above)
Comment (WunderBart) [reply packages/calypso-e2e/src/element-helper.ts]: Updated in https://github.com/Automattic/wp-calypso/pull/69180/commits/13a2017cc15a5e39fe5f64093e97406268faf21b

**Pull request merged**